src/include/migraphx/shape_for_each.hpp

```diff
@@ -34,14 +34,14 @@ inline namespace MIGRAPHX_INLINE_NS {
 /**
  * Iterates the given function over the indices from the shape in order.
  */
-template <class F>
+template <class F, class Vec = std::vector<std::size_t>, class Idx = size_t>
 void shape_for_each(const migraphx::shape& s, F f)
 {
-    // Ensure calls to f use const ref to vector
-    auto call = [&f](const std::vector<std::size_t>& i) { f(i); };
     std::vector<std::size_t> indices(s.lens().size());
+    auto const& index_const_ref = indices;
```

Comment:
wouldn't it be same if it was `const auto& index_const_ref` ?

Reply:
Sure. Thanks. Will change it..

Reply:
Done